### PR TITLE
Version 3.2.5

### PR DESCRIPTION
**Pull request by konraddysput**
## Version 3.2.5
- Added `BacktraceClient` Initialization method that allows developer to intialize Backtrace integration without adding game object to game scene.
- Fixed invalid `meta` file for iOS integration for Unity 2019.2.13f1.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Backtrace Unity Release Notes
 
+## Version 3.2.5
+- Added `BacktraceClient` Initialization method that allows developer to intialize Backtrace integration without adding game object to game scene.
+- Fixed invalid `meta` file for iOS integration for Unity 2019.2.13f1.
+- Backtrace client allows to configure report sampling. By default report sampling is equal to 0.01 - which means 1% of all unhandled exceptions will be skipped.
```
Comment (sbahra) [inline CHANGELOG.md:6]: The expectation here is that 1% of unhandled exceptions will be reports. Worth also clarifying that it is random sampling.

```
- Enables a new random sampling mechanism for unhandled exceptions.
```

File: Runtime/BacktraceDatabase.cs [hunk 2]
```diff
@@ -534,8 +534,9 @@ private bool ValidateDatabaseSize()
             //check how many records are stored in database
             //remove in case when we want to store one more than expected number
             //If record count == 0 then we ignore this condition
-            if (BacktraceDatabaseContext.Count() + 1 > DatabaseSettings.MaxRecordCount && DatabaseSettings.MaxRecordCount != 0 && !BacktraceDatabaseContext.RemoveLastRecord())
-            {
+            var noMoreSpaceForReport = BacktraceDatabaseContext.Count() + 1 > DatabaseSettings.MaxRecordCount && DatabaseSettings.MaxRecordCount != 0;
```
Comment (sbahra) [inline Runtime/BacktraceDatabase.cs:537]: Why this new semantic? Previously we used to remove the last record.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Backtrace Unity Release Notes
 
+## Version 3.2.5
+- Added `BacktraceClient` Initialization method that allows developer to intialize Backtrace integration without adding game object to game scene.
+- Fixed invalid `meta` file for iOS integration for Unity 2019.2.13f1.
+- Backtrace client allows to configure report sampling. By default report sampling is equal to 0.01 - which means 1% of all unhandled exceptions will be skipped. 
+- HTTP communication messages improvements.
```
[the remark this answers is not in the document]
Comment (konraddysput) [reply CHANGELOG.md:7]: Added information about failure message.

File: Editor/BacktraceConfigurationEditor.cs [hunk 4]
```diff
@@ -75,6 +75,11 @@ public override void OnInspectorGUI()
                     serializedObject.FindProperty("DestroyOnLoad"),
                     new GUIContent(BacktraceConfigurationLabels.LABEL_DESTROY_CLIENT_ON_SCENE_LOAD));
 
+
```
[the remark this answers is not in the document]
Comment (konraddysput) [reply Editor/BacktraceConfigurationEditor.cs:78]: Removed new line

File: Runtime/BacktraceClient.cs [hunk 5]
```diff
@@ -601,15 +676,36 @@ private void CaptureUnityMessages()
...
+        {
+#if UNITY_EDITOR
+            return false;
+#else
+            if (!Configuration || Configuration.Sampling == 0)
+            {
+                return false;
+            }
+            return UnityEngine.Random.Range(0f, 1f) < Configuration.Sampling;
+#endif
```
[the remark this answers is not in the document]
Comment (konraddysput) [reply Runtime/BacktraceClient.cs:706]: Added a clarification about sampling usage in editor mode. Also this information is available in method summary.

File: Runtime/BacktraceDatabase.cs (hunk 2 above)
Comment (konraddysput) [reply Runtime/BacktraceDatabase.cs:537]: We're removing this error later - if it's necessary. Previously with this behavior we will override an error with deduplication counter.

File: CHANGELOG.md (hunk 1 above)
Comment (konraddysput) [reply CHANGELOG.md]: Clarify this information in changelog and readme. Thanks

**Pull request merged**